webui/module/Restore/src/Restore/Form/RestoreForm.php

```diff
@@ -599,137 +435,95 @@ public function __construct($restore_params = null, $clients = null, $filesets =
+            'name' => 'form-submit',
+            'type' => 'button',
+            'options' => array(
+                'label' => _('Restore')
+            ),
+            'attributes' => array(
+                //'value' => _('Restore'),
+                'id' => 'btn-form-submit',
+                'class' => 'btn btn-primary',
+                // will be enabled by JS
+                'disabled' => true
+            )
+        );
+        $this->add($form_submit);
 
-    /**
-     *
-     */
-    private function getJobList()
-    {
-        $selectData = array();
-        if (!empty($this->jobs)) {
-            foreach ($this->jobs as $job) {
-                $selectData[$job['jobid']] = $job['jobid'] . " - " . $job['name'];
-            }
-        }
-        return $selectData;
     }
 
     /**
-     *
+     * Convert an Bareos result array so that it is usable as Form Options array.
+     * Here it sets key and value to the "name" field of the Bareos array.
      */
-    private function getBackupList()
+    private function getNameOptionsMap($source)
     {
         $selectData = array();
-        if (!empty($this->backups)) {
-            foreach ($this->backups as $backup) {
-                switch ($backup['level']) {
-                    case 'I':
-                        $level = "Incremental";
-                        break;
-                    case 'D':
-                        $level = "Differential";
-                        break;
-                    case 'F':
-                        $level = "Full";
-                        break;
-                }
-                $selectData[$backup['jobid']] = "(" . $backup['jobid']  . ") " . $backup['starttime'] . " - " . $backup['name'] . " - " . $level;
+        if (!empty($source)) {
+            foreach ($source as $i) {
+                $selectData[$i["name"]] = $i["name"];
             }
         }
+        ksort($selectData);
         return $selectData;
     }
 
     /**
      *
      */
-    private function getJobIds()
-    {
-        return $this->jobids;
-    }
-
-    /**
-     *
-     */
-    private function getClientList()
+    private function getJobList()
     {
         $selectData = array();
-        if (!empty($this->clients)) {
-            foreach ($this->clients as $client) {
-                $selectData[$client['name']] = $client['name'];
+        if (!empty($this->jobs)) {
+            foreach ($this->jobs as $job) {
+                $selectData[$job['jobid']] = $job['jobid'] . " - " . $job['name'];
             }
         }
-        ksort($selectData);
         return $selectData;
     }
 
     /**
      *
      */
-    private function getFilesetList()
+    private function getBackupList()
     {
         $selectData = array();
-        if (!empty($this->filesets)) {
-            foreach ($this->filesets as $fileset) {
-                $selectData[$fileset['name']] = $fileset['name'];
+        foreach ($this->backups as $backup) {
```

Comment:
The constructor sets `this->backups` to `null`  by default, so we probably need the `!empty($this->backups)` check here.

Reply:
done